cmd/thanos/bucket.go

```diff
@@ -234,15 +235,18 @@ func registerBucketLs(m map[string]setupFunc, root *kingpin.CmdClause, name stri
 			}
 		}
 
-		if err := bkt.Iter(ctx, "", func(name string) error {
-			id, ok := block.IsBlockDir(name)
-			if !ok {
-				return nil
-			}
+		metas, _, metaFetchErr := fetcher.Fetch(ctx)
+		// For partial view allow adding new blocks at least.
+		if metaFetchErr != nil && metas == nil {
```

Comment:
In Go we stick to `err` variable name for errors (: Even if there is already an `err` variable - that's fine as long as you don't need differentiate those.

Also, do we need to check if `metas` == nil? In Go there is a pattern that if `err` is not nil, caller has to ignore rest of the arguments as they are incorrect... due to error (: